panoramapublic/src/org/labkey/panoramapublic/view/combinationModInfo.jsp

```diff
@@ -0,0 +1,327 @@
+            xtype: 'combo',
+            name: 'unimodId' + cbIdx,
+            itemId: 'unimodId' + cbIdx,
+            fieldLabel: "Unimod Modification " + cbIdx,
+            allowBlank: false,
+            editable : true,
+            queryMode : 'local',
+            forceSelection: true, // restrict the selected value to one of the values in the list
+            displayField: 'displayName',
+            valueField: 'id',
+            value: cbIdx === 1 ? <%=form.getUnimodId1() != null ? form.getUnimodId1() : null %> :
+                                 <%=form.getUnimodId2() != null ? form.getUnimodId2() : null %>,
+            store: createStore(),
+            labelWidth: 160,
+            width: 600,
+            labelStyle: 'background-color: #E0E6EA; padding: 5px;',
+            listeners: {
+                scope: this,
+                select: function (combo, records){
+                    const record = records[0];
+                    if (cbIdx === 1) selectedUnimod1 = record.data;
+                    else selectedUnimod2 = record.data;
+                    updateFormulaDiff();
+                },
+                change: function(combo, newValue) {
+                    const store = combo.getStore();
+                    store.clearFilter(true);
+                    store.addFilter(createDataSourceFilter(newValue));
+                }
+            }
+        });
+    }
+
+    const alertCls = 'alert';
+    const alertInfoCls = 'alert-info';
+    const alertWarnCls = 'alert-warning';
+
+    function updateFormulaDiff() {
+
+        let totalFormula = new Formula();
+        totalFormula = addUnimod(totalFormula, selectedUnimod1);
+        totalFormula = addUnimod(totalFormula, selectedUnimod2);
+        const diffFormula = modFormula.subtractFormula(totalFormula);
+        const formulaBalanced = diffFormula.isEmpty();
+
+        let html = '';
+        html += selectedUnimod1 ? selectedUnimod1["formula"] : span('---');
+        html += span('+');
+        html += selectedUnimod2 ? selectedUnimod2["formula"] : span('---');
+        html += span('=');
+        html += totalFormula.getFormula();
+        if (formulaBalanced) {
+            html += '<span style="color:green;margin-left:10px;" class="fa fa-check-circle"></span>';
+        }
+        else {
+            html += span(' (Difference: ' + diffFormula.getFormula() + ')');
+            html += '<span style="color:red;" class="fa fa-times-circle"></span>';
+        }
+
+        const el = formPanel.getComponent("formulaDiff");
+        if (el != null)
+        {
+            if (!el.hasCls(alertCls)) el.addCls(alertCls);
+
+            el.removeCls(formulaBalanced ? alertWarnCls : alertInfoCls);
+            el.addCls(formulaBalanced ? alertInfoCls : alertWarnCls);
+            el.update(html);
+        }
+    }
+
+    function span(str) {
+        return '<span style="margin: 0 10px 0 10px;font-weight:bold;">' + str + '</span>';
```

Comment:
Perhaps also known to be safe, but probably also wise to HTML encode str here

Reply:
Done.